src/qpx_mmc_css.cpp

```diff
@@ -0,0 +1,1909 @@
+  drive->cmd[11] = 0;
+  if ((drive->err = drive->cmd.transport(READ, drive->rd_buf, len))) {
+    if (!drive->silent)
+      sperror("READ_DISC_REGIONS", drive->err);
+    return drive->err;
+  }
+#if 0
+    if (!drive->silent) {
+        printf("READ_DISC_REGIONS data: ");
+        for(i=0; i<len ; i++) printf(" %02X",drive->rd_buf[i] & 0xFF);
+        printf("\n");
+    }
+#endif
+  drive->media.dvdcss.protection = drive->rd_buf[4];
+  drive->media.dvdcss.regmask = drive->rd_buf[5];
+
+  if (!drive->media.dvdcss.protection) {
+    //	printf("DVD is NOT protected\n");
+    return 0;
+  } else {
+    /*
+    switch (drive->media.dvdcss.protection) {
+            case 0x01:
+                    printf("DVD is CSS-protected\n");
+                    break;
+            case 0x02:
+                    printf("DVD is CPRM-protected\n");
+                    break;
+            default:
+                    printf("Unknown DVD protection shceme!\n");
+                    break;
+    }*/
+
+    //		printf("Disc regions       : ");
+    if (drive->media.dvdcss.regmask != 0xFF) {
+      for (i = 0; i < 8; i++)
+        if (!((drive->media.dvdcss.regmask >> i) & 1)) {
+          //	printf("%d",i+1);
+          drive->rpc.region = i + 1;
+        }
+      //			printf("\n");
+      //		} else {
+      //			printf("Invalid region mask!\n");
+    }
+  }
+
+  return 0;
+}
+
+int read_disc_key(drive_info *drive, unsigned char DK[DVD_DISCKEY_SIZE]) {
+  int len = 2052; // 4 (header) + 2048 (key)
+  if (!(drive->rd_capabilities & DEVICE_DVD) ||
+      !(drive->capabilities & CAP_DVD_CSS))
+    return -1;
+
+  drive->cmd[0] = MMC_READ_DVD_STRUCTURE;
+  drive->cmd[7] = 0x02;
+  drive->cmd[8] = len >> 8;   //  -|
+  drive->cmd[9] = len & 0xFF; //  -- transfer length
+  drive->cmd[10] = drive->media.dvdcss.agid & 0xC0;
+  drive->cmd[11] = 0;
+  if ((drive->err = drive->cmd.transport(READ, drive->rd_buf, len))) {
+    if (!drive->silent)
+      sperror("READ_DISC_KEY", drive->err);
+    return drive->err;
+  }
+
+  memcpy(DK, drive->rd_buf + 4, 2048);
+  return 0;
+}
+
+int report_key(drive_info *drive, unsigned char key_class,
+               unsigned char key_format, int len, unsigned int lba) {
+  if (!(drive->rd_capabilities & DEVICE_DVD) ||
+      !(drive->capabilities & CAP_DVD_CSS))
+    return -1;
+
+  drive->cmd[0] = MMC_REPORT_KEY;
+
+  drive->cmd[2] = (lba >> 24) & 0xFF;
+  drive->cmd[3] = (lba >> 16) & 0xFF;
+  drive->cmd[4] = (lba >> 8) & 0xFF;
+  drive->cmd[5] = lba & 0xFF;
+
+  drive->cmd[7] = key_class;
+  drive->cmd[8] = len >> 8;
+  drive->cmd[9] = len & 0xFF;
+  drive->cmd[10] = (drive->media.dvdcss.agid & 0xC0) | (key_format & 0x3F);
+  drive->cmd[11] = 0;
+  if ((drive->err = drive->cmd.transport(READ, drive->rd_buf, len))) {
+    if (!drive->silent)
+      sperror("MMC REPORT KEY", drive->err);
+    return drive->err;
+  }
+  return 0;
+}
+
+int get_rpc_state(drive_info *drive) {
+  //	int len=8;
+  int i;
```

Comment:
Codacy found an issue: [The scope of the variable 'i' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)